sphinx/builders/latex/theming.py

```diff
@@ -0,0 +1,118 @@
+"""
+    sphinx.builders.latex.theming
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    Theming support for LaTeX builder.
+
+    :copyright: Copyright 2007-2019 by the Sphinx team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+import configparser
+from os import path
+from typing import Dict
+
+from sphinx.application import Sphinx
+from sphinx.config import Config
+from sphinx.errors import ThemeError
+from sphinx.locale import __
+from sphinx.util import logging
+
+logger = logging.getLogger(__name__)
+
+
+class Theme:
+    """A set of LaTeX configurations."""
+
+    def __init__(self, name: str) -> None:
+        self.name = name
+        self.docclass = name
+        self.wrapperclass = name
```

Comment:
I wonder wrapperclass is a part of LaTeX theme. Actually it is used in sphinxhowto.cls and sphinxmanual.cls. But other themes may not follow the implementation, I think. It would be better to separate styles from sphinxhowto.cls and sphinxmanual.cls into indivisual .sty file.

Reply:
Of course wrapperclass is an heritage of Sphinx own way: it uses a base class such as report or article. Then some document layout provided via Sphinx (title page for example) is configured in a way depending taking into account base class. But in the context of a user document class, it does not make much sense to have same mechanism: either the user will base a LaTeX class on book or report or article or whatever. So the user wishing to theme LaTeX should only have to provide own class and matched template. Possibly the gist of sphinxmanual/sphinxhowto should be moved as you say into .sty files, say sphinxreportstyle.sty and sphinxarticlestyle.sty (thus, we drop manual/howto in naming) and sphinxjsreportstyle.sty etc... A user is the free to  import or not one of those in a custom class.